src/itemvalue.ts

```diff
@@ -76,15 +76,19 @@ export class ItemValue extends Base implements ILocalizableOwner, IShortcutText
       return result;
     };
   }
-  public static setData(items: Array<ItemValue>, values: Array<any>) {
+
+  /**
+   * Resets the input array and fills it with values from the values array
+   */
+  public static setData(items: Array<ItemValue>, values: Array<any>, type?:string): void {
     items.length = 0;
-    for (var i = 0; i < values.length; i++) {
-      var value = values[i];
-      var item: ItemValue;
+    for (let i = 0; i < values.length; i++) {
+      const value = values[i];
+      let item: ItemValue;
```

Comment:
I would changed it as the following:
      const value = values[i];
      const itemType = !!value && typeof value.getType === "function" ? value.getType() : (type ?? "itemvalue");
      const item = Serializer.createClass(itemType);